receiver/jaegerreceiver/trace_receiver_test.go

```diff
@@ -101,6 +106,41 @@ func TestReception(t *testing.T) {
 	}
 }
 
+func TestPortsNotOpen(t *testing.T) {
+	// an empty config should result in no open ports
+	config := &Configuration{}
+
+	sink := new(exportertest.SinkTraceExporter)
+
+	jr, err := New(context.Background(), config, sink, zap.NewNop())
+	assert.NoError(t, err, "should not have failed to create a new receiver")
+	defer jr.StopTraceReception()
+
+	mh := receivertest.NewMockHost()
+	err = jr.StartTraceReception(mh)
+	assert.NoError(t, err, "should not have failed to start trace reception")
+
+	// there is a race condition here that we're ignoring.
+	//   this test may occasionally pass incorrectly, but it will not fail incorrectly
```

Comment:
Does this realistically ever lose the race and verify anything? Unless you introduce a deliberate pause my guess would be this almost always executed before StartTraceReception has a chance to listen. I could be wrong, let me know if you tested for this.

Side note: we should probably allow components to asynchronously signal end of Start so that tests can wait for that.

Reply:
I added this test and verified it failed consistently before adding the fix.  Given that it was consistently failing I chose not to add a pause to not add anymore time to CI.  I would be glad to add a pause (or even remove the test) if you'd like.

I agree on your side note.  There's a number of tests where we have a "WaitForPort()" call.  A direct signal would be much cleaner.

Reply:
If it fails consistently or even most of the time then it is good enough for now. We can revisit the test when we have better signalling of completion from components to host. Please add a TODO to the comment to reflect this discussion.